pkg/utils/tokenizer/compat_vllm.go

```diff
@@ -0,0 +1,113 @@
+/*
+Copyright 2024 The Aibrix Team.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package tokenizer
+
+import (
+	"context"
+	"fmt"
+	"time"
+)
+
+const (
+	// Default values for VLLMTokenizerConfig
+	defaultVLLMTimeout    = 30 // seconds
+	defaultVLLMMaxRetries = 3
+)
+
+// VLLMTokenizer implements both Tokenizer and ExtendedTokenizer interfaces
+// It acts as a proxy to forward tokenization requests to vLLM server
+// Deprecated: This is a compatibility wrapper. Use NewRemoteTokenizer instead.
+type VLLMTokenizer struct {
+	config *VLLMTokenizerConfig
+	remote RemoteTokenizer
+}
+
+// NewVLLMTokenizer creates a new vLLM tokenizer instance
+// Deprecated: Use NewRemoteTokenizer with RemoteTokenizerConfig instead
+func NewVLLMTokenizer(config VLLMTokenizerConfig) (*VLLMTokenizer, error) {
```

Comment:
I did not get the part, why is there a need for NewVLLMTokenizer and not directly use adapter_vllm via remote tokenizer?